phantazm-zombies/src/main/java/com/github/phantazmnetwork/zombies/equipment/gun/audience/AudienceProvider.java

```diff
@@ -0,0 +1,12 @@
+package com.github.phantazmnetwork.zombies.equipment.gun.audience;
+
+import net.kyori.adventure.audience.Audience;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Optional;
+
+public interface AudienceProvider {
```

Comment:
@FunctionalInterface

Reply:
also this is just a Supplier<Optional<? extends Audience>>, any reason to have it be its own thing

Reply:
less typing, multiple implentations